### ISSUE TITLE
Hook stdout output isn't printed or logged

### ISSUE THREAD

**Issue by stephenostermiller** [certbot/certbot#4167]
I had created a certificate using a command like this:

    certbot certonly --staging --agree-tos --webroot -w /var/www/example.com/ -d example.com

I created a test script for post renewal

    #!/bin/sh
    set -e 
    echo "RENEWED_LINEAGE: $RENEWED_LINEAGE"
    echo "RENEWED_DOMAINS: $RENEWED_DOMAINS"

And I run certbot to force a renewal:

    $ sudo certbot --staging renew --force-renewal --renew-hook /home/steveo/config/scripts/lets-encrypt-renew-hook.sh 
    -------------------------------------------------------------------------------
    Processing /etc/letsencrypt/renewal/example.com.conf
    -------------------------------------------------------------------------------
    
    -------------------------------------------------------------------------------
    new certificate deployed without reload, fullchain is
    /etc/letsencrypt/live/example.com/fullchain.pem
    -------------------------------------------------------------------------------
    
    Congratulations, all renewals succeeded. The following certs have been renewed:
    /etc/letsencrypt/live/example.com/fullchain.pem (success)

But my renew hook script never ran.   If it had, I would expect to see it print the environment variables.

**Comment by bmw** [comment]
This is a duplicate of #3947. `--renew-hook` is only run when renewing certificates with `certbot renew`, however, we're going to deprecate the flag in favor of `--deploy-hook` which will always be run when obtaining a new certificate.

**Comment by stephenostermiller** [comment]
I was running certbot renew and it never got called.

**Comment by bmw** [comment]
I'm sorry. I missed that in the latter part of your post. I should have read more closely.

What version of `certbot` are you using? Certbot should provide more output than what you posted above. Running the same commands on `master`, I get the following output for `renew`:
```
sudo certbot --staging renew --force-renewal --renew-hook ./test.sh
Saving debug log to /var/log/letsencrypt/letsencrypt.log

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/example.com.conf
-------------------------------------------------------------------------------
Renewing an existing certificate
Performing the following challenges:
http-01 challenge for example.com
Waiting for verification...
Cleaning up challenges
Generating key (2048 bits): /etc/letsencrypt/keys/0001_key-certbot.pem
Creating CSR: /etc/letsencrypt/csr/0001_csr-certbot.pem
Running renew-hook command: ./test.sh

-------------------------------------------------------------------------------
new certificate deployed without reload, fullchain is
/etc/letsencrypt/live/example.com/fullchain.pem
-------------------------------------------------------------------------------

Congratulations, all renewals succeeded. The following certs have been renewed:
  /etc/letsencrypt/live/example.com/fullchain.pem (success)
```
As for not seeing the output of `renew-hook`, only `stderr` output is printed. If you change your test script to:
```
#!/bin/sh
set -e 
echo "RENEWED_LINEAGE: $RENEWED_LINEAGE" >&2
echo "RENEWED_DOMAINS: $RENEWED_DOMAINS" >&2
```
or
```
#!/bin/sh
set -e 
echo "RENEWED_LINEAGE: $RENEWED_LINEAGE" >> out.txt
echo "RENEWED_DOMAINS: $RENEWED_DOMAINS" >> out.txt
```
you should see the output either in the terminal or in `./out.txt` respectively.

**Comment by stephenostermiller** [comment]
I'm using the version in the Ubuntu 16.10 repositories:

    $ certbot --version
    certbot 0.8.1

If certbot supresses STDOUT from the script that would certainly explain why I didn't see it.   I can change my script, but that seems like really dumb behavior.   At the very least it should be documented as a big gotcha.

**Comment by bmw** [comment]
We can potentially look into changing the behavior.

If a developer makes this change, we'd want to log `stdout` to `logger.info`.

**Comment by bmw** [comment]
Fixed in #4702.
